src/gc/impl/conservative/gc.d

```diff
@@ -2078,6 +2078,135 @@ struct Gcx
     ToScanStack!(ScanRange) toscanPrecise;
 
     /**
+     * Finds the offset between an index and the next set bit, up to our max.
+     *
+     * Returns: The offset between index and the next bit,
+     *          or -1 if no bits are set between index and max
+     */
+    int bitScanForwardOffset(size_t* bitArray, size_t index, size_t max) nothrow
```

Comment:
> They should be changing each time bitScanForwardOffset is called. 

Yes, but the previous call almost always knows the answer for the next by just adding the offset.

Previous versions of the precise GC had this main marking loop before the GC was refactored to not recurse:

https://github.com/rainers/druntime/blob/gcx_precise2/src/gc/gc.d#L2699

Adding `bsf` to the increment might make it more efficient.